Gems/Atom/Tools/ShaderManagementConsole/Code/Source/Window/ShaderManagementConsoleWindow.cpp

```diff
@@ -41,6 +41,25 @@ namespace ShaderManagementConsole
         SetDockWidgetVisible("Inspector", false);
 
         OnDocumentOpened(AZ::Uuid::CreateNull());
+        this->setContextMenuPolicy(Qt::CustomContextMenu);
+        this->connect(this, &QTableWidget::customContextMenuRequested, this, &ShaderManagementConsoleWindow::ShowContextMenu);
+    }
+
+    void ShaderManagementConsoleWindow::ShowContextMenu(const QPoint& pos)
+    {
+        QMenu contextMenu(tr("Context menu"), this);
+        QAction* action = new QAction(QString(tr("Open recent...")), this); // for example.
+        connect(
+            action,
+            &QAction::triggered,
+            this,
+            []()
+            {
+                
+            });
+        contextMenu.addAction(action);
+
+        contextMenu.exec(mapToGlobal(pos));
```

Comment:
I figured it'd be convenient to have a small set of "File" menu actions repeated directly from the blank space context menu. Not implemented yet.

Reply:
The main file menu has opened recent functionality built into it. I believe you can require a pointer to that menu and display it as a pop up menu here.

Reply:
I did that, it works very poorly because of some 1 or 2 second delay in the construction of that menu, requiring to pass the mouse cursor a second time over the submenu to see it pop.